### ISSUE TITLE
reconfigure virtual machines

### ISSUE THREAD

**Issue by rismoney** [terraform-providers/terraform-provider-vsphere#385]
Is it possible to change Terraform's behavior such that, changes to existing VMs (which require a power cycle), the powercycle can be deferred to a later time? 

I suppose this would be a tweak of shutdown_wait_timeout/force_power_off handling or similar.  When you make a change in the GUI, essentially the change is deferred until a subsequent restart.  With TF, at the time of `apply` the node will restart.    This becomes tricky whereby orchestration and configuration become interweaved.

**Comment by vancluever** [comment]
Hey @rismoney, the current behaviour is actually there because there can be several operations that actually require the VM to be powered down in order to be changed. If they are changed via API an error is issued.

I'm not too sure if the behaviour is different from GUI to GUI, but a lot of the settings that do require a restart were determined by actually looking at the new HTML5 vSphere client on a powered on VM and checking to see if that setting was greyed out or not. What GUI are you using and do you have an example of a setting that you are trying to modify?

**Comment by rismoney** [comment]
I heard from the sysadmin - he was using the Windows fat client.  I  believe your summary of the HTML5 is accurate.   

I suppose, I'd envision some sort of terraform plan indication of a pending powercycle, (much like FNR)

**Comment by vancluever** [comment]
Hey @rismoney, this is actually something that is under discussion as hashicorp/terraform#16624.

When/if it becomes a thing, we definitely will add some additional messages to the provider so that these details show up in the diff!

Going to close this now as it seems like we got a consensus here on what is going on and how we can solve it in the long run, but given that this support is not in TF at this point in time it will probably be a while before we can revisit this.
